mne_bids/inspect.py

```diff
@@ -81,12 +93,24 @@ def _inspect_raw(*, bids_path, l_freq, h_freq, verbose=None):
     # closed, our dialog box will pop up, asking whether to save changes.
     def _handle_close(event):
         mne_raw_fig = event.canvas.figure
+        # Bads alterations are only transferred to `inst` once the figure is
+        # closed; Annotation changes are immediately reflected in `inst`
         new_bads = mne_raw_fig.mne.info['bads'].copy()
-
-        _save_bads_if_changed(old_bads=old_bads,
-                              new_bads=new_bads,
-                              bids_path=bids_path,
-                              verbose=verbose)
+        new_annotations = mne_raw_fig.mne.inst.annotations.copy()
+
+        if not new_annotations:
+            # Ensure it's not an empty list, but an empty set of Annotations.
+            new_annotations = (mne_raw_fig.mne.inst.copy()
+                               .set_annotations(None)
+                               .annotations
+                               .copy())
```

Comment:
I don't get why you need this? making a copy of potentially a 4GB raw object to only look at annotations ?

Reply:
Resolved in bcb277b